### DISCUSSION TITLE
How to modify query in a resource view page ?

### DISCUSSION THREAD

**Discussion by amoktar** [filamentphp/filament#10806] [Help]
Hello guys.
How can I modify query in a resource view page 

As an example here:
I'm using a view page for a resource `ProductResource`
https://filamentphp.com/docs/3.x/panels/resources/viewing-records

I'm using global scope in the model `Product`. 

Now I want to disable the global scope in the resource view page

Thanks

**Comment by leandrocfe** [comment]
You can modify the record using the mount method in the ViewPage file

```php
public function mount(int|string $record): void
{
    parent::mount($record);

    $this->record = ...
}
```

**Reply by amoktar** [reply]
Thanks @leandrocfe ,
Actaully I'm trying to chain `withoutGlobalScopes` to the query . 
The model/record is not loaded because of the global scope

**Reply by leandrocfe** [reply]
Where? I didn't understand what you need

**Reply by amoktar** [reply]
Thanks again. 
I'm using global scope with my `Product` model. The record is not loaded and getting 404 becase of the global scope .
So, I want to disable the global scope in the filament resource view page

**Comment by mazenbassiouni** [comment]
you can rewrite the mount function as follows:

```
public function mount(int|string $record): void
{
        $this->record = $this->getModel()::withoutGlobalScopes()->findOrFail($record);

        $this->authorizeAccess();

        if (! $this->hasInfolist()) {
            $this->fillForm();
        }
}
```
